Create stacktrace documentation (#6556)
highlight/highlight: docs-content/general/6_product-features/2_error-monitoring/enhancing-errors-with-github.md

---
title: Enhancing Errors with GitHub
slug: enhancing-errors-with-github
createdAt: 2023-09-07T16:07:50.273Z
updatedAt: 2023-09-07T16:07:50.273Z
---

> Highlight has to capability to enhance your backend errors using GitHub. On the frontend, errors are enhanced using sourcemaps uploaded to Highlight. Some backend

Reviewer: ```suggestion
Highlight has the capability to enhance your backend errors using GitHub (errors on the frontend are enhanced using sourcemaps). Some backend
```

Is there a sourcemap doc we can link here in case someone lands on the wrong doc?

Author: Yep! Will add that link in here

> languages will give us context of a stacktrace, but for most compiled languages, there is no access to this. Using GitHub, these stacktraces are able to get this

Reviewer: ```suggestion
languages will give us context of a stacktrace, but for most compiled languages, Highlight doesn't have access to this. With our GitHub integration, when Highlight enhances a stacktrace, we are are able to add stacktrace context, 
```

> context, as well as other enhancements such as link directly to a file in GitHub and attributing errors to file changes.

Reviewer: ```suggestion
, as well as other enhancements such as "link to a file" and file attribution.
```

In order to turn on GitHub Enhancements, 3 actions need to be completed for your project:
<ol>
  <li>1. Add the GitHub Integration to Highlight</li>
  <li>2. Create a service via the SDK</li>
  <li>3. Link your service to a GitHub repo</li>
</ol>

> 

Reviewer: Maybe services should be first?

More information on the GitHub Integration can be found at [GitHub Integration](../../7_integrations/github-integration.md).

## Create a service via the SDK
Services are created to group your logs, errors, and traces by the process that is running the code. Having a service can make it helpful to decipher
which process caused an error, especially in code paths shared by multuple processes. They can also be used also filters for logs and traces.

In order to create a new service, a service name must be added to your SDK. Reference the [SDK start up guides](../../../getting-started/1_overview.md) for more help.

For example, in Golang, it would look something like:
```
highlight.SetProjectID("<YOUR_PROJECT_ID>")
highlight.Start(
    highlight.WithServiceName("my-app"),
    highlight.WithServiceVersion("git-sha"),
)
defer highlight.Stop()
```

<b>Note:</b> There is also a service version that is provided in the example above. It is not necessary to enable GitHub enhancements, but is recommended that this be the current GIT SHA of the deployed code to use the most accurate files. If not provided, Highlight will fallback to your current default branch (e.g. main) GIT SHA.

> 

Reviewer: Maybe we can screenshot what a service would look like once they've added it? and where to find that part of the app?

Reviewer: ah you already did that later.

## Link your service to a GitHub repo

> Once a service is created, it should be visible in your project settings, under the "Services" table. The last step to enable stacktrace enhancements is to link your service to a GitHub repo, the one that should be used to enhance your errors. In addition to linking the repo, there are two fields to set configure any file path mappings from your deployment process that would be needed to correctly find a file on GitHub.
> 

Reviewer: ```suggestion
Once a service is created, it should be visible in your project settings, under the "Services" table. The last step to enable stacktrace enhancements is to link your service to its respective GitHub repo. In addition to linking the repo, there are two required fields to configure file path mappings from your deployment process to the correct file in GitHub.

```

> 1. <b>Build path prefix</b> - This path prefix was added in your deployment process, and is the path in your server that contains your files. This path is not found in GitHub,

Reviewer: ```suggestion
1. <b>Build path prefix</b> - This path prefix represents a path added in your deployment process, and is also the path in your server that contains your files. After removing this path (and possibly adding something else), you should be able to point this string to a github file.
```

> and should be removed when fetching the file.

Reviewer: ```suggestion
```

> 2. <b>GitHub path prefix</b> - This path prefix was removed in your deployment process, and should be prepended to your files in order to correctly find the file on GitHub.

Reviewer: ```suggestion
2. <b>GitHub path prefix</b> - This path prefix is a string that can be appended to the front of the stacktrace path, and will be prepended to your files in order to correctly find the file on GitHub.
```

An example:
<ol>

>     <li>1. An error received has a stacktrace that points to the file `/build/main.go`.</li>

Reviewer: ```suggestion
    <li>1. An error received has a stacktrace path `/build/main.go`.</li>
```

>     <li>2. The GitHub repo was selected to be the Highlight repo.</li>

Reviewer: ```suggestion
    <li>2. The GitHub repo was selected to be the Highlight repo.</li>
```
link the repo here?

>     <li>3. Highlight's deployment process, move's all files out of the `/backend` directory and into a `/build` directory.</li>

Reviewer: ```suggestion
    <li>3. Since Highlight's deployment process moves all files out of the `/backend` directory and into the `/build` directory,  we would set "Build prefix path" to `/backend` and GitHub prefix path to `/backend`</li>
```

---
title: Enhancing Errors with GitHub
slug: enhancing-errors-with-github
createdAt: 2023-09-07T16:07:50.273Z
updatedAt: 2023-09-07T16:07:50.273Z
---

> Highlight has the capability to enhance your backend errors using GitHub (errors on the frontend are enhanced using [sourcemaps](./sourcemaps.md)). Some backend
> languages will give us context of a stacktrace, but for most compiled languages, Highlight doesn't have no access to this. With our GitHub integration, Highlight
> is able to enhance a stacktrace with context, as well as other enhancements such as "link to a file" and attribution to a code change.

Reviewer: ```suggestion
Highlight has the capability to enhance your backend errors using GitHub (errors on the frontend are enhanced using [sourcemaps](./sourcemaps.md)). With our GitHub integration, Highlight
is able to enhance a stacktrace with additional lines from your original code base, as well as other enhancements such as "link to a file" and attribution to a code change.
```

In order to turn on GitHub Enhancements, 3 actions need to be completed for your project:
<ol>
  <li>1. Create a service via the SDK</li>
  <li>2. Add the GitHub Integration to Highlight</li>
  <li>3. Link your service to a GitHub repo</li>
</ol>

## Create a service via the SDK

> Services are created to group your logs, errors, and traces by the process that is running the code. Having a service can make it helpful to decipher

Reviewer: ```suggestion
Services are created to group your logs, errors, and traces by the process that is running the code. Having a service can make it helpful to decipher
```
i'm not sure if process is a good word here...

which process caused an error, especially in code paths shared by multuple processes. They can also be used also filters for logs and traces.

In order to create a new service, a service name must be added to your SDK. Reference the [SDK start up guides](../../../getting-started/1_overview.md) for more help.

For example, in Golang, it would look something like:
```
highlight.SetProjectID("<YOUR_PROJECT_ID>")
highlight.Start(
    highlight.WithServiceName("my-app"),
    highlight.WithServiceVersion("git-sha"),
)
defer highlight.Stop()
```

> <b>Note:</b> There is also a service version that is provided in the example above. It is not necessary to enable GitHub enhancements, but is recommended that this be the current GIT SHA of the deployed code to use the most accurate files. If not provided, Highlight will fallback to your current default branch (e.g. main) GIT SHA.

Reviewer: ```suggestion
<b>Note:</b> There is also a service version that is provided in the example above. This is not necessary to enable GitHub enhancements, but is recommended that this be the current GIT SHA of the deployed code to use the most accurate files. If not provided, Highlight will fallback to your current default branch (e.g. main) GIT SHA.
```
